test/clj/rems/poller/test_email.clj

```diff
@@ -81,22 +78,39 @@
    "handler" {:commonName "Hannah Handler"
               :email "hannah@handler.com"}})
 
-(defn events-to-emails [events]
-  (let [application (-> (reduce model/application-view nil events)
-                        (model/enrich-with-injections {:get-workflow get-workflow
-                                                       :get-catalogue-item get-catalogue-item
-                                                       :get-form-template get-form-template
-                                                       :get-license get-license
-                                                       :get-user get-nothing
-                                                       :get-users-with-role get-nothing
-                                                       :get-attachments-for-application get-nothing}))]
-    (with-redefs [rems.config/env (assoc rems.config/env :public-url "http://example.com/")
-                  rems.db.users/get-user-attributes get-user-attributes
-                  user-settings/get-user-settings (constantly {:language :en})]
-      (mapv #(sort-emails (#'rems.poller.email/event-to-emails-impl % application)) events))))
+(defn email-recipient [email]
+  (or (:to email) (:to-user email)))
+
+(defn sort-emails [emails]
+  (sort-by email-recipient emails))
+
+(defn email-recipients [emails]
+  (set (mapv #(or (:to %) (:to-user %)) emails)))
```

Comment:
minor: could use email-recipient function here

Reply:
oh right that's what I meant to do